include/WCSimGenerator_Radioactivity.hh

```diff
@@ -24,133 +31,82 @@ class WCSimGenerator_Radioactivity
 		
 		G4ThreeVector GetRandomVertex(G4int tSymNumber);
-		G4double GetMeanActivity() { return fIntegral; }
-		G4double GetIDVolume() { return (fR_max*fR_max*TMath::Pi()*fZ_max*2.); }
+		G4double GetMeanActivity()		{ return fConcentrationID; }
+		G4double GetMeanActivityReco()	{ return fConcentrationReco; }
+		G4double GetIDVolume()			{ return (fR_max*fR_max*TMath::Pi()*fZ_max*2.); }
 		
 		
 	private:
 	
-		static G4double ZFit_R210(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double ZFit_R190(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double ZFit_R170(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double ZFit_R140(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double ZFit_R070(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double ZFit_R040(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double ZFit_R020(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double ZFit_R000(G4double x, G4double Lambda, G4double BinConversion);
-
-		static G4double R2Fit_ZpM(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double R2Fit_ZmM(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double R2Fit_Z16(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double R2Fit_Z13(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double R2Fit_Z10(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double R2Fit_Z00(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double R2Fit_Z14p(G4double x, G4double Lambda, G4double BinConversion);
-		static G4double R2Fit_Z16p(G4double x, G4double Lambda, G4double BinConversion);
+		// Diffusion functions
+		static double DiffusionZ(double x, double origin, double factor);
+		static double DiffusionR(double x, double origin, double factor);
+		
+		// Concentration functions
+		static double RadonFormulaR(double* val, double* par);
+		static double RadonFormulaZ(double* val, double* par);
 		
 		// Function
 		TF2* thRnFunction;
 		G4double fIntegral;
+		G4double fIntegralReco;
+		
+		G4double fConcentrationID;
+		G4double fConcentrationReco;
 		
 		void SetScenario(G4int iScenario);
-		static G4double RadonFormula(G4double *val, G4double *par);
+		static double RadonFormula(double *val, double *par);
 
 		WCSimDetectorConstruction*      myDetector;
 		
 		// Constant	
 		G4int    fScenario;
 		static G4double fRnDiffusion_Coef;
+		static G4double fRnLambda_Global;
 		G4double fRnLambda;
 		
 		static G4double fRn_PerPMT;
 		
-		static G4double fRnSK_Center;
-		static G4double fRnSK_Bottom;
+		// Parameter
+		G4double CONC_BOTTOM;
+		G4double CONC_CENTER;
+		G4double CONC_INTERMEDIATE;
+	
+		G4double CONC_MIDDLE;
+		G4double CONC_INT_R2_6;
+		G4double CONC_INT_R2_7;
+		G4double CONC_INT_R2_8;
+		G4double CONC_INT_R2_9;
 		
 		static G4double fRn_Border;
 		
+		static G4double fHK_Z_min;
```

Comment:
I see from later in the code that you're doing some scaling for "current detector geometry" relative to SK. Therefore, since WCSim can run multiple geometries (e.g. HK, SK, IWCD) the HK part of the variable name has the potential to be confusing
So something like `fCurrDet_Z_min` would be preferable

Reply:
Done